### PR TITLE
Member create modal

### PR DESCRIPTION
**Pull request by deepsidhu85**
## What does this PR do and why?
_Describe in detail what your merge request does and why._

This PR adds functionality to add a member to a group/project using a modal and to use turbo_stream.

## Screenshots or screen recordings
_Screenshots are required for UI changes, and strongly recommended for all other pull requests._

image

image

image

## How to set up and validate locally
_Numbered steps to set up and validate the change are strongly suggested._

1. Login as any user
2. Visit a groups members page
3. Click the `Add New Member` button
4. A modal should pop up for you to select a user and their access level
5. Select a user and their access level
6. Click the `Add member to group` button
7. The table and pagination should update to include this user
8. Verify in the network tab of the browser dev tools that the request to add a member to a group is sent via turbo_stream
9. Verify the same functionality for a project
10. Verify in the network tab of the browser dev tools that the request to add a member to a project is sent via turbo_stream

Note: If trying to add a member to a subgroup or a project where the user also has membership in one of it's ancestors, with a lower access level you should see an error in the modal such as:

image

## PR acceptance checklist
This checklist encourages us to confirm any changes have been analyzed to reduce risks in quality, performance, reliability, security, and maintainability.

- [X] I have evaluated the [PR acceptance checklist](https://phac-nml.github.io/irida-next/docs/development/development_processes/code_review#acceptance-checklist) for this PR.

### CODE REVIEW

**Comment by joshsadam** [comment]
This is still doing a full page fetch on the project add new member 
image

**Comment by deepsidhu85** [comment]
> This is still doing a full page fetch on the project add new member, it is working properly on the groups though. image

I can't seem to replicate it..these are what my network tab looks like for both group members new and project members new

image

image

**Comment by joshsadam** [comment]
I was actually wrong before (not in your favour though), the response to both is returning a full HTML page.  Look at the network tab you showed me above, click on the first call named `new` and look at the response, it's the full page.  This is turbo's normal action and gives you the entire page and then just replaces what it wants.  What I think you are trying to do is to get the partial that just has the dialogue contents in it.

**Comment by deepsidhu85** [comment]
> I was actually wrong before (not in your favour though), the response to both is returning a full HTML page on both project and groups. Look at the network tab you showed me above, click on the first call named `new` and look at the response, it's the full page. This is turbo's normal action and gives you the entire page and then just replaces what it wants. What I think you are trying to do is to get the partial that just has the dialogue contents in it.

As discussed the purpose of this PR was to open the create modal via a server call and not have the new page load on a separate page. The request to create the member is what is being sent via a turbo stream request

File: app/components/viral/dialog/header_component.html.erb [hunk 1]
```diff
@@ -1,7 +1,7 @@
 <%= render Viral::BaseComponent.new(tag: "div", **system_arguments) do %>
-  <h3 class="dialog--title">
+  <h1 class="dialog--title">
```
Comment (joshsadam) [inline app/components/viral/dialog/header_component.html.erb:2]: **nitpick:** Why did you change this to an h1 tag? There should only ever be 1 h1 tag on a page, and the title of a dialogue is probably not that.  The flowbite documentation has it as an h3.

**Review by joshsadam — CHANGES_REQUESTED** [review]
Overall, from what I see, this looks good.  I would still like to see the dialogue being rendered using a turbo_stream instead of a full-page request.  Good work Deep.

File: app/components/viral/dialog/header_component.html.erb (hunk 1 above)
Comment (joshsadam) [reply app/components/viral/dialog/header_component.html.erb:2]: Apparently, you are correct.  Thanks for updating this.

**Comment by deepsidhu85** [comment]
> Overall, from what I see, this looks good. I would still like to see the dialogue being rendered using a turbo_stream instead of a full-page request. Good work Deep.

Thanks! I updated the modals to be rendered via turbo_stream in 4ba6e11798a68b9286ba3c74ce734d70ba5e1eb7

**Review by ericenns — APPROVED** [review]
Looks good to me! I tested it all out and it is working as expected

**Review by joshsadam — APPROVED** [review]

**Pull request merged**